### ISSUE TITLE
Cassandra example can't possibly work after v1b3 conversion

### ISSUE THREAD

**Issue by thockin** [GoogleCloudPlatform/kubernetes#7949]
https://github.com/GoogleCloudPlatform/kubernetes/blob/master/examples/cassandra/java/src/io/k8s/cassandra/KubernetesSeedProvider.java#L67

It's is using v1beta3 but referencing endpoints.endpoints, which was a v1beta1 field.  The conversion was just changing the API version, which is insufficient.  You have to actually fix the code.

@caesarxuchao @nikhiljindal

**Comment by caesarxuchao** [comment]
I'll take care of it

**Comment by thockin** [comment]
Thanks!

**Comment by pires** [comment]
Same happens to `hazelcast` example.

```
$ kubectl get endpoints
NAME            ENDPOINTS
hazelcast       10.1.0.2:5701
(...)

 curl http://10.100.0.1:80/api/v1beta3/endpoints/hazelcast
{
  "kind": "Status",
  "apiVersion": "v1beta3",
  "metadata": {},
  "status": "Failure",
  "message": "the server could not find the requested resource",
  "reason": "NotFound",
  "details": {},
  "code": 404
}

$ curl http://10.100.0.1:80/api/v1beta1/endpoints/hazelcast
{
  "kind": "Endpoints",
  "id": "hazelcast",
  "uid": "80490202-f67a-11e4-a69a-080027aa505e",
  "creationTimestamp": "2015-05-09T18:37:52Z",
  "selfLink": "/api/v1beta1/endpoints/hazelcast?namespace=default",
  "resourceVersion": 523,
  "apiVersion": "v1beta1",
  "namespace": "default",
  "protocol": "TCP",
  "endpoints": [
    "10.1.0.2:5701"
  ],
  "targetRefs": [
    {
      "endpoint": "10.1.0.2:5701",
      "targetRef": {
        "kind": "Pod",
        "namespace": "default",
        "name": "hazelcast-2faw2",
        "uid": "8264dc6c-f67a-11e4-a69a-080027aa505e",
        "resourceVersion": "522"
      }
    }
  ],
  "subsets": [
    {
      "addresses": [
        {
          "IP": "10.1.0.2",
          "targetRef": {
            "kind": "Pod",
            "namespace": "default",
            "name": "hazelcast-2faw2",
            "uid": "8264dc6c-f67a-11e4-a69a-080027aa505e",
            "resourceVersion": "522"
          }
        }
      ],
      "ports": [
        {
          "port": 5701,
          "protocol": "TCP"
        }
      ]
    }
  ]
}
```

**Comment by pires** [comment]
Question, what do I/we need to do?

**Comment by caesarxuchao** [comment]
Hi @pires, in v1beta3, the URL becomes

```
http://10.100.0.1:80/api/v1beta3/namespaces/deafult/endpoints/hazelcast
```

Could you give it a try? Thanks.

#7731

**Comment by pires** [comment]
@caesarxuchao I have fixed it, but had to revert to `v1beta1` in the URL. Now I get it, thanks. Want me to PR a fix for this as well?

**Comment by pires** [comment]
The `Endpoints` class no longer fits the API so need to fix that as well.

**Comment by caesarxuchao** [comment]
Do you mean for the Cassandra example? I'm writing the PR now. Thank you : )

On Saturday, May 9, 2015, Paulo Pires <notifications@github.com> wrote:

> @caesarxuchao <https://github.com/caesarxuchao> I have fixed it, but had
> to revert to v1beta1 in the URL. Now I get it, thanks. Want me to PR a
> fix for this as well?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/issues/7949#issuecomment-100532054>
> .
>


-- 
Regards,
Chao Xu

**Comment by pires** [comment]
@caesarxuchao are you on irc.freenode.net? Let's talk. `ppires` here.

**Comment by caesarxuchao** [comment]
Sorry, just saw your message. I sent you several messages on irc. Not sure if I used the correct command. "/msg ppires blahblah"

**Comment by caesarxuchao** [comment]
fixed by #8025
